app/routes/items/new.js

```diff
@@ -69,78 +54,98 @@ export default AuthorizeRoute.extend({
     });
   },
 
-  async setupController(controller, model) {
+  setupController(controller, model) {
     this._super(controller, model);
+    const store = this.get("store");
+    this.initializeController();
+
+    if (this.get("newItemRequest")) {
+      this.initializeDimensions();
+      this.manageSubformDetails();
+      this.setUpPackageImage();
+      window.localStorage.setItem("isSelectLocationPreviousRoute", false);
+    }
+  },
 
-    controller.set("inventoryNumber", this.get("inventoryNumber"));
-    controller.set("displayInventoryOptions", false);
-    controller.set("autoGenerateInventory", true);
-    controller.set("inputInventory", false);
+  initializeController() {
+    const controller = this.controller;
+    if (!controller.get("inventoryNumber")) {
+      controller.send("autoGenerateInventoryNumber");
```

Comment:
Since the inventory number generation is done on the backend, I'm pretty sure this could lead to asynchronous drama.

You can await the inventory number creation, set it, and then proceed